src/main.ts

```diff
@@ -521,12 +521,13 @@ function generateDecode(
 ): FunctionSpec {
   // create the basic function declaration
   let func = FunctionSpec.create('decode')
-    .addParameter('reader', 'Reader@protobufjs/minimal')
+    .addParameter('input', TypeNames.unionType('Uint8Array', 'Reader@protobufjs/minimal'))
     .addParameter('length?', 'number')
     .returns(fullName);
 
   // add the initial end/message
   func = func
+    .addStatement('const reader = input instanceof Uint8Array ? new Reader(input) : input')
```

Comment:
In Node, `Buffer` is a sub-class of `Uint8Array`, and the browser doesn't have `Buffer` -- just `Uint8Array`, so this `instanceof` check seems ok to cover those use-cases.

Reply:
Nice. Wouldn't mind that as an in-source comment. I'd forgotten Buffer subclasses Uint8Array.